samples/authenticator/src/main/java/org/forgerock/authenticator/sample/view/activity/PushNotificationActivity.java

```diff
@@ -30,18 +52,239 @@ public void onCreate(Bundle savedInstanceState) {
+                                finish();
+                            }
+                        });
+                    }
+                });
+            }
+        });
+
+        Button choice2Button = findViewById(R.id.choice2);
+        choice2Button.setText(String.valueOf(choices[1]));
+        choice2Button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                notification.accept(String.valueOf(choices[1]), new FRAListener<Void>() {
+                    @Override
+                    public void onSuccess(Void result) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, String.format(getString(R.string.push_notification_approval), "approved"), Toast.LENGTH_LONG).show();
+                                finish();
+                            }
+                        });
+                    }
+
+                    @Override
+                    public void onException(Exception e) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, e.getLocalizedMessage(), Toast.LENGTH_LONG).show();
+                                finish();
+                            }
+                        });
+                    }
+                });
+            }
+        });
+
+        Button choice3Button = findViewById(R.id.choice3);
+        choice3Button.setText(String.valueOf(choices[2]));
+        choice3Button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                notification.accept(String.valueOf(choices[2]), new FRAListener<Void>() {
+                    @Override
+                    public void onSuccess(Void result) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, String.format(getString(R.string.push_notification_approval), "approved"), Toast.LENGTH_LONG).show();
+                                finish();
+                            }
+                        });
+                    }
+
+                    @Override
+                    public void onException(Exception e) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, R.string.notification_error_network_failure_message, Toast.LENGTH_LONG).show();
```

Comment:
Update the toast message to display the actual error - i.e. e.getLocalizedMessage()